src/Bicep.Core/Analyzers/Linter/Rules/NoHardcodedEnvironmentUrlsRule.cs

```diff
@@ -36,56 +44,56 @@ public override void Configure(IConfigurationRoot config)
         public override string FormatMessage(params object[] values)
             => string.Format("{0} Found this disallowed host: \"{1}\"", this.Description, values.First());
 
+        private Regex CreateDisallowedHostRegex() =>
+            new Regex(string.Join('|', this.DisallowedHosts), RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
Programmatically combining Regex patterns is very difficult - I would recommend either instead building a dedicated regex for each disallowed host (if regex matching is required), or just using simple string search (if regex matching is not required).
Some examples of why this is difficult:
* If I add a host named `(abc` to the config, it'll break this logic because the regex combining rule doesn't escape special characters.
* The config you've added to the default config `schema.management.azure.com` is technically more permissive - as e.g. `schema-management.azure.com` will also match it (since `.` means "any character" in regex). I think the correct regex to use in the config would be `schema\.management\.azure\.com` (escaping the `.` to make it only match a literal `.`).

Reply:
I've got an update for this and I want to do speed tests on individual regex and string matching to inform this decision.

Reply:
Regarding perf: The combined regex is Many times faster in every scenario over multip regex and string.contains type operations